zmq.hpp

```diff
@@ -2224,6 +2224,53 @@ class monitor_t
         on_monitor_started();
     }
 
+#if ZMQ_VERSION_MAJOR >= 4
+    bool get_event(zmq_event_t& eventMsg, std::string& address, zmq::recv_flags flags = zmq::recv_flags::none)
```

Comment:
I am not sure about the signature of this method. What about 
```
std::optional<std::pair<zmq_event_t, std::string>> get_event(zmq::recv_flags flags = zmq::recv_flags::none)
```
(This requires C++17 this way, but we can also do this with the fallback to `detail::trivial_optional` as done for other types)

This removes the ambiguity of whether the original value of `eventMsg` and `address` are used and in what cases they are modified.

Reply:
Ok, I'm not very familiar with std::optional and will try to implement this way